numba/tests/test_unicode.py

```diff
@@ -337,6 +345,69 @@ def test_find(self, flags=no_pyobj_flags):
                                  cfunc(a, substr),
                                  "'%s'.find('%s')?" % (a, substr))
 
+    def test_rfind(self):
+        pyfunc = rfind_usecase
+        cfunc = njit(pyfunc)
+
+        str_examples = [
+            'abcde',
+            'aabbccddbb',
+            'a22abd2',
+            'abc' * 15,
+            'aBcDefBc',
+            '1223456557',
+            ' ',
+            'ab*cf+*bc',
+            'abc aade',
+            'a555bcgf5555' ]
+
+        substr_examples = [
+            'e',
+            'bb',
+            ' ',
+            '*+',
+            '55',
+            'B' ]
+
+        for s in str_examples:
+            for sub_str in substr_examples:
+                self.assertEqual(pyfunc(s, sub_str),
+                                 cfunc(s, sub_str),
+                                 "'%s'.rfind('%s')?" % (s, sub_str))
+
+    def test_rfind_with_startend(self):
+        pyfunc = rfind_with_startend_usecase
+        cfunc = njit(pyfunc)
+
+        str_examples = [
+            'abcde',
+            'aabbccddbb',
+            'a22abd2',
+            'abc' * 15,
+            'aBcDefBc',
+            '1223456557',
+            ' ',
+            'ab*cf+*bc',
+            'abc aade',
+            'a555bcgf5555' ]
+
+        substr_examples = [
+            'e',
+            'bb',
+            ' ',
+            '*+',
+            '55',
+            'B' ]
+
+        begin = 4
```

Comment:
it would be good to test some variations in `begin` and `end` including the use of `None`.